universal-application-tool-0.0.1/app/auth/UatProfileAdapter.java

```diff
@@ -55,7 +56,36 @@ public UatProfileAdapter(
       return Optional.empty();
     }
 
+    Optional<UatProfile> existingProfile = profileUtils.currentUserProfile(context);
     OidcProfile profile = (OidcProfile) oidcProfile.get();
+    // Check if we already have a profile in the database for this user.
+    Optional<Applicant> existingApplicant =
+        ApplicantRepository.INSTANCE
+            .lookupApplicant(profile.getEmail())
+            .toCompletableFuture()
+            .join();
+
+    // Now we have a three-way merge situation.  We might have
+    // 1) an applicant in the database,
```

Comment:
nit: can you put the corresponding variable names next to each list item just so its super clear?